frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/EditTab.tsx

```diff
@@ -1,102 +1,104 @@
-import React, { useState } from 'react';
+import React, { createRef } from 'react';
+import {
+  StudioAlert,
+  StudioButton,
+  StudioErrorMessage,
+  StudioSpinner,
+  usePrevious,
+} from '@studio/components';
 import { useTranslation } from 'react-i18next';
-import { StudioDeleteButton, StudioErrorMessage } from '@studio/components';
-import { AddManualOptionsModal } from './AddManualOptionsModal';
-import { OptionListSelector } from './OptionListSelector';
-import { OptionListUploader } from './OptionListUploader';
-import { OptionListEditor } from './/OptionListEditor';
+import { useUpdate } from 'app-shared/hooks/useUpdate';
 import { useComponentErrorMessage } from '../../../../../../hooks';
-import type { IGenericEditComponent } from '../../../../componentConfig';
+import {
+  handleOptionsChange,
+  updateComponentOptions,
+  isOptionsModifiable,
+  isOptionsIdReferenceId,
+} from '../utils/optionsUtils';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useOptionListIdsQuery } from '../../../../../../hooks/queries/useOptionListIdsQuery';
 import type { SelectionComponentType } from '../../../../../../types/FormComponent';
+import type { IGenericEditComponent } from '../../../../componentConfig';
+import { OptionListSelector } from './OptionListSelector';
+import { OptionListUploader } from './OptionListUploader';
+import { OptionListEditor } from './OptionListEditor';
 import classes from './EditTab.module.css';
 
-type EditOptionChoiceProps = Pick<
+type EditTabProps = Pick<
   IGenericEditComponent<SelectionComponentType>,
   'component' | 'handleComponentChange'
 >;
 
-export function EditTab({
-  component,
-  handleComponentChange,
-}: EditOptionChoiceProps): React.ReactElement {
-  const initialComponentHasOptionList: boolean = !!component.optionsId || !!component.options;
-  const [componentHasOptionList, setComponentHasOptionList] = useState<boolean>(
-    initialComponentHasOptionList,
-  );
+export function EditTab({ component, handleComponentChange }: EditTabProps): React.ReactElement {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionListIds, status } = useOptionListIdsQuery(org, app);
+  const previousComponent = usePrevious(component);
+  const dialogRef = createRef<HTMLDialogElement>();
   const errorMessage = useComponentErrorMessage(component);
 
-  return (
-    <>
-      {componentHasOptionList ? (
-        <SelectedOptionList
-          setComponentHasOptionList={setComponentHasOptionList}
-          component={component}
-          handleComponentChange={handleComponentChange}
+  useUpdate(() => {
+    if (!previousComponent.options && !!component.options) dialogRef.current.showModal();
```

Comment:
```suggestion
    if (isInitialOptionsAdded(previousComponent.options, component.options)) dialogRef.current.showModal();
```

and add something like this below the component;

const isInitialOptionsAdded = (previousOptions: OptionsList, currentOptions: OptionsList) => !previousOptions && !!currentOptions;

Reply:
Or in the optionsUtilsFile 😄